mender-update/daemon/state_machine.hpp

```diff
@@ -107,6 +109,293 @@ class StateMachine {
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			idle_leave_inv_(
+				loop,
+				script_executor::State::Idle,
+				script_executor::Action::Leave,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			sync_enter_deployment_(
+				loop,
+				script_executor::State::Sync,
+				script_executor::Action::Enter,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			sync_enter_inventory_(
+				loop,
+				script_executor::State::Sync,
+				script_executor::Action::Enter,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			sync_leave_(
+				loop,
+				script_executor::State::Sync,
+				script_executor::Action::Leave,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			sync_leave_download_(
+				loop,
+				script_executor::State::Sync,
+				script_executor::Action::Leave,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			sync_error_(
+				loop,
+				script_executor::State::Sync,
+				script_executor::Action::Error,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			download_enter_(
+				loop,
+				script_executor::State::Download,
+				script_executor::Action::Enter,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path,
+				Context::kUpdateStateDownload),
+			download_leave_(
+				loop,
+				script_executor::State::Download,
+				script_executor::Action::Leave,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			download_leave_save_provides(
+				loop,
+				script_executor::State::Download,
+				script_executor::Action::Leave,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			download_error_(
+				loop,
+				script_executor::State::Download,
+				script_executor::Action::Error,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			install_enter_(
+				loop,
+				script_executor::State::ArtifactInstall,
+				script_executor::Action::Enter,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path,
+				Context::kUpdateStateArtifactInstall),
+			install_leave_(
+				loop,
+				script_executor::State::ArtifactInstall,
+				script_executor::Action::Leave,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			install_error_(
+				loop,
+				script_executor::State::ArtifactInstall,
+				script_executor::Action::Error,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			install_error_rollback_(
+				loop,
+				script_executor::State::ArtifactInstall,
+				script_executor::Action::Error,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path),
+			reboot_enter_(
+				loop,
+				script_executor::State::ArtifactReboot,
+				script_executor::Action::Enter,
+				retry_interval,
+				artifact_script_path,
+				rootfs_script_path,
+				Context::kUpdateStateArtifactReboot,
+				true),
```

Comment:
You added `true` here in the latest update. Was it intentional, or were you just a bit quick when adding it to `rollback_reboot_enter` (where it indeed should be)?

Reply:
Uhh! Good catch!